src/plugins/jni/README.md

```diff
@@ -47,6 +48,25 @@ and run:
 
     sudo ldconfig
 
+### Java prerequisites on macOS ###
+
+macOS includes an old apple specific version of java, based on 1.6. However, for the jni plugin version 1.8 of Java is required, so either the openjdk or the oracle jdk has to be installed.
+
+Please install oracle's jdk8 via their provided installer. After that, you have to set the JAVA_HOME environment variable to the folder where the jdk is installed, usually like
+
+	export JAVA_HOME="/Library/Java/JavaVirtualMachines/jdk1.8.0_112.jdk/Contents/Home/"
+
+If it should still not find the correct jni version, or says the jni version is not 1.8, then it most likely still searches in the wrong directory for the jni header file. You can try clearing your build folder and configuring it again from scratch.
```

Comment:
yes, this is enough, updated documentation according to that. Note that this will also reset any changes you have make with e.g. ccmake instead. 
also restructured this and moved to a troubleshooting section